src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,29 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+
+        public static string GetMeaningfulParentName(SyntaxNode current, SemanticModel model)
```

Comment:
That's fine. Two points: if you remove the method name from the message, this method goes away. If not, it needs a comment to explain what it does.